pkg/admissioncontroller/webhooks/admission/auditpolicy/admission_test.go

```diff
@@ -0,0 +1,536 @@
+apiVersion: audit.k8s.io/v1
+kind: Policy
+rules:
+  - level: RequestResponse
+    resources:
+    - group: ""
+      resources: ["pods"]
+  - level: Metadata
+    resources:
+    - group: ""
+      resources: ["pods/log"]
+`
+	)
+
+	BeforeEach(func() {
+		logger = logzap.New(logzap.WriteTo(GinkgoWriter))
+		testEncoder = &json.Serializer{}
+
+		ctrl = gomock.NewController(GinkgoT())
+		mockReader = mockclient.NewMockReader(ctrl)
+
+		var err error
+		decoder, err = admission.NewDecoder(kubernetes.GardenScheme)
+		Expect(err).NotTo(HaveOccurred())
+
+		handler = auditpolicy.New(logger)
+		Expect(inject.APIReaderInto(mockReader, handler)).To(BeTrue())
+		Expect(admission.InjectDecoderInto(decoder, handler)).To(BeTrue())
+
+		request = admission.Request{}
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("#ValidateAuditPolicyApiGroupVersionKind", func() {
+		var kind = "Policy"
+
+		It("should return false without error because of version incompatibility", func() {
+			incompatibilityMatrix := map[string][]schema.GroupVersionKind{
+				"1.10.0": {
+					auditv1.SchemeGroupVersion.WithKind(kind),
+				},
+				"1.11.0": {
+					auditv1.SchemeGroupVersion.WithKind(kind),
+				},
+			}
+
+			for shootVersion, gvks := range incompatibilityMatrix {
+				for _, gvk := range gvks {
+					ok, err := auditpolicy.IsValidAuditPolicyVersion(shootVersion, &gvk)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(ok).To(BeFalse())
+				}
+			}
+		})
+
+		It("should return true without error because of version compatibility", func() {
+			compatibilityMatrix := map[string][]schema.GroupVersionKind{
+				"1.10.0": {
+					auditv1alpha1.SchemeGroupVersion.WithKind(kind),
+					auditv1beta1.SchemeGroupVersion.WithKind(kind),
+				},
+				"1.11.0": {
+					auditv1alpha1.SchemeGroupVersion.WithKind(kind),
+					auditv1beta1.SchemeGroupVersion.WithKind(kind),
+				},
+				"1.12.0": {
+					auditv1alpha1.SchemeGroupVersion.WithKind(kind),
+					auditv1beta1.SchemeGroupVersion.WithKind(kind),
+					auditv1.SchemeGroupVersion.WithKind(kind),
+				},
+				"1.13.0": {
+					auditv1alpha1.SchemeGroupVersion.WithKind(kind),
+					auditv1beta1.SchemeGroupVersion.WithKind(kind),
+					auditv1.SchemeGroupVersion.WithKind(kind),
+				},
+				"1.14.0": {
+					auditv1alpha1.SchemeGroupVersion.WithKind(kind),
+					auditv1beta1.SchemeGroupVersion.WithKind(kind),
+					auditv1.SchemeGroupVersion.WithKind(kind),
+				},
+				"1.15.0": {
+					auditv1alpha1.SchemeGroupVersion.WithKind(kind),
+					auditv1beta1.SchemeGroupVersion.WithKind(kind),
+					auditv1.SchemeGroupVersion.WithKind(kind),
+				},
+			}
+
+			for shootVersion, gvks := range compatibilityMatrix {
+				for _, gvk := range gvks {
+					ok, err := auditpolicy.IsValidAuditPolicyVersion(shootVersion, &gvk)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(ok).To(BeTrue())
+				}
+			}
+		})
+
+		It("should return false with error because of not valid semver version", func() {
+			shootVersion := "1.ab.0"
```

Comment:
Could you also try with something like `1.12.3-dev`.
We had some troubles with such versions (with the OCI images and GKE versions) and it will be nice to have a test for it.

Reply:
I added a test - can you check if I understood you correctly?